### DISCUSSION TITLE
How to correctly structure a datapack?

### DISCUSSION THREAD

**Discussion by Aiyron** [ejektaflex/Bountiful#188] [Modpack Help]
### What happened?

Good morning,
To start sorry I only have two choices and "bug" is the most logical.

I'm looking to add my own decrees to the mod via a datapack but I don't have the info on how to link the mod to the datapack

i don't know anything about datapack but minecraft doesn't say anything i've seen about it and neither does the mod wiki

I created a .mcmeta (pack, pack format, description)
(my mod adds roses and I wanted to test with it to get an idea of how the mod works).

I tried to put at the beginning of my test in "rose.json" the classic pool of the mod (objs and rews) without success.

I created three .json:
rose.json
rose_objs.json
rose_rews.json

in "rose.json" I put as objective "rose_objs" (even with the .json between the " " it does not work)

in "rose_objs.json" I put three possible lime blue and purple roses
in "rose_rews.json" I put as reward "aiycoin:coin_stone" (one coin) imagining that a single reward in the .json can work (because I don't have the info) even if several rewards don't change Nothing

so after my 8h test impossible to have my personalized decree I don't understand

Am I missing something in my .mcmeta?
should i use a particular name for the datapack (missing info)
can i create my own objs or am i forced to use the basic ones
how to link the datapack to the mod (minecraft detects my datapack well)

the three "rose" .json files are the most classic nothing to show special I tried with what your wiki explains (everything is correctly written)

help :'(

if you need more info please let me know because i really need it for my modpack and removing the mod would be sad.

Version 3.3.1 - 1.16.4
forge 1.6.5 (36.2.34)

From google translate if mistakes are readable

### Mod Version

1.16.4 - 3.3.1

### Forge Version

1.16.5 36.2.34

### Relevant log output

"pack.mcmeta"
--------------------------------- start of file
{ 
    "pack" :  { 
        "pack_format" :  6 , 
        "description" :  "Quêtes et récompenses de bountiful personnalisé" 
    } 
}
--------------------------------- end of file
"rose.json"
--------------------------------- start of file
{
	"spawnsInBoard": true,
	"objectivePools": [
	"rose_objs.json"
],
"rewardPools": [
"rose_rews.json"
]
}
--------------------------------- end of file
"rose_objs.json"
--------------------------------- start of file
{
	"content": [
{
  "type": "item",
  "content": "aiyserv:rose_lime",
  "amountRange": {
    "min": 8,
    "max": 12
  },
  "unitWorth": 25,
  "weight": 3000
},
{
  "type": "item",
  "content": "aiyserv:rose_purple",
  "amountRange": {
    "min": 2,
    "max": 8
  },
  "unitWorth": 25,
  "weight": 1000
},
{
  "type": "item",
  "content": "aiyserv:rose_blue",
  "amountRange": {
    "min": 3,
    "max": 45
  },
  "unitWorth": 25,
  "weight": 2000
}
]
}
--------------------------------- end of file
"rose_rews"
--------------------------------- start of file
{
  "content": [
    {
      "type": "item",
      "content": "aiycoin:coin_stone",
      "amountRange": {
        "min": 1,
        "max": 3
      },
      "unitWorth": 1200,
      "weight": 6000
    }
]}

**Comment by ejektaflex** [comment]
Could you attach the datapack as a zip? Then I can take a look at it.
Thanks!

**Comment by Aiyron** [comment]
[bountiful.zip](https://github.com/ejektaflex/Bountiful/files/11040526/bountiful.zip)
all is here but nothing new (without the subfolder nothing change)

note: i use subfolder for test not the zip so normal if the folder is in the zip i just zipped now

**Comment by ejektaflex** [comment]
I think that your folder structure is a bit off - you'll want a folder structure like this:

* `mypack`
  * `pack.mcmeta`
  * `mypack`
    * `pools`
      * `mypack`
        * `my_obj_file.json`
        * `my_rew_file.json`
    * `decrees`
      * `mypack`
        * `my_decree_file.json`

I hope that helps! Data pack structures are pretty confusing. It should follow the same outline as my files here:
https://github.com/ejektaflex/Bountiful/tree/1.16/src/main/resources/data

**Reply by Annijang** [reply]
Can you make a beta version of datapack please? I would like to check it out, thanks.

**Comment by ejektaflex** [comment]
I'm marking this discussion as closed - for future wandering eyes, an example pack for newer versions can be found at: https://modrinth.com/datapack/bountiful-examplepack
